### PR TITLE
Specification design

### PR DESCRIPTION
**Pull request by mpenciak**

### CODE REVIEW

**Comment by mpenciak** [comment]
The runner and DSL don't work, but I wanted to lock in the spec design before going forward.

File: Main.lean [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import LSpec.Runner
+
+def main : IO UInt32 := runner
```
Comment (arthurpaulino) [inline Main.lean:3]: Can we just explicitly put the runner code here?

Comment (arthurpaulino) [inline Main.lean:3]: Can we just explicitly put the runner code here?

File: LSpec/Syntax.lean [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    `(doElem| results := results.concat (ExampleOf.fromDescrParam (equals $l $r) $s ()).run )
+  | `(doElem| it $s so $f never equals $b check $a) => 
+    `(doElem| results := results.concat (ExampleOf.fromDescrParam (alwaysEquals $f $b) $s $a).run )
+  | `(testSuite| test_suite $ts*) => 
+      `(def testName : IO UInt32 := do
+          let mut results : List String := []
+          $[$ts:doElem]*
+          for result in results do
+            IO.println result
+          return 0)
```
Comment (arthurpaulino) [inline LSpec/Syntax.lean:20]: I think we can indeed delete this. We'll need to think more deeply about how we want to set tests to run in separate files

Comment (arthurpaulino) [inline LSpec/Syntax.lean:20]: I think we can indeed delete this. We'll need to think more deeply about how we want to set tests to run in separate files

File: LSpec/Config.lean [hunk 3]
```diff
@@ -0,0 +1,3 @@
+-- TODO: Config data structure more complicated than this
+structure Config where
+  temp : Bool
```
Comment (arthurpaulino) [inline LSpec/Config.lean:3]: Do we need this? Configuring the behavior of an imported package should be done with a proper option via `set_option`, I believe. So this probably wouldn't be the way

Comment (arthurpaulino) [inline LSpec/Config.lean:3]: Do we need this? Configuring the behavior of an imported package should be done with a proper option via `set_option`, I believe. So this probably wouldn't be the way

File: Main.lean (hunk 1 above)
Comment (mpenciak) [reply Main.lean:3]: Yeah feel free to put it here.

File: LSpec/Config.lean (hunk 3 above)
Comment (mpenciak) [reply LSpec/Config.lean:3]: This is just a placeholder, so you can delete it for now. I had it in place to eventually support some of the configurations like they have here

https://hspec.github.io/options.html

If you think set_option can subsume this kind of functionality then we won't need it.

**Review by arthurpaulino — APPROVED** [review]

**Pull request merged**